packages/firebase_ui_firestore/lib/src/query_builder.dart

```diff
@@ -449,57 +449,363 @@ class FirestoreListView<Document> extends FirestoreQueryBuilder<Document> {
+        keyboardDismissBehavior: keyboardDismissBehavior,
+        restorationId: restorationId,
+        clipBehavior: clipBehavior,
+      );
+    },
+  );
+
+  /// Shows a separator between list items just as in [ListView.separated]
+  FirestoreListView.separated({
+    Key? key,
+    required Query<Document> query,
+    required FirestoreItemBuilder<Document> itemBuilder,
+    int pageSize = 10,
+    FirestoreLoadingBuilder? loadingBuilder,
+    FirestoreErrorBuilder? errorBuilder,
+    FirestoreEmptyBuilder? emptyBuilder,
+    required IndexedWidgetBuilder separatorBuilder,
+    Axis scrollDirection = Axis.vertical,
+    bool reverse = false,
+    ScrollController? controller,
+    bool? primary,
+    ScrollPhysics? physics,
+    bool shrinkWrap = false,
+    EdgeInsetsGeometry? padding,
+    ChildIndexGetter? findChildIndexCallback,
+    bool addAutomaticKeepAlives = true,
+    bool addRepaintBoundaries = true,
+    bool addSemanticIndexes = true,
+    double? cacheExtent,
+    DragStartBehavior dragStartBehavior = DragStartBehavior.start,
+    ScrollViewKeyboardDismissBehavior keyboardDismissBehavior =
+        ScrollViewKeyboardDismissBehavior.manual,
+    String? restorationId,
+    Clip clipBehavior = Clip.hardEdge,
+  }) : super(
+    key: key,
+    query: query,
+    pageSize: pageSize,
+    builder: (context, snapshot, _) {
+      if (snapshot.isFetching) {
+        return loadingBuilder?.call(context) ??
+            const Center(child: CircularProgressIndicator());
+      }
+
+      if (snapshot.hasError && errorBuilder != null) {
+        return errorBuilder(
+          context,
+          snapshot.error!,
+          snapshot.stackTrace!,
+        );
+      }
+
+      if (snapshot.docs.isEmpty && emptyBuilder != null) {
+        return emptyBuilder(context);
+      }
+
+      return ListView.separated(
+        itemCount: snapshot.docs.length,
+        itemBuilder: (context, index) {
+          final isLastItem = index + 1 == snapshot.docs.length;
+          if (isLastItem && snapshot.hasMore) snapshot.fetchMore();
+
+          final doc = snapshot.docs[index];
+          return itemBuilder(context, doc);
+        },
+        separatorBuilder: separatorBuilder,
+        scrollDirection: scrollDirection,
+        reverse: reverse,
+        controller: controller,
+        primary: primary,
+        physics: physics,
+        shrinkWrap: shrinkWrap,
+        padding: padding,
+        findChildIndexCallback: findChildIndexCallback,
+        addAutomaticKeepAlives: addAutomaticKeepAlives,
+        addRepaintBoundaries: addRepaintBoundaries,
+        addSemanticIndexes: addSemanticIndexes,
+        cacheExtent: cacheExtent,
+        dragStartBehavior: dragStartBehavior,
+        keyboardDismissBehavior: keyboardDismissBehavior,
+        restorationId: restorationId,
+        clipBehavior: clipBehavior,
+      );
+    },
+  );
+
+  /// Shows a load indicator when the last item is reached
+  FirestoreListView.withLoadingIndicator({
```

Comment:
is there a way to have this configurable via constructor property instead of a separate constructor? While `FirestoreListView.separated` plays nicely with `ListView.separated`, `withLoadingIndicator` and `separatedWithLoadingIndicator` seem like an overkill.

Reply:
Thanks for the review!
I'll look into it and get back to you.